### PR TITLE
React Refactor: UNSAFE methods `alerts-view.tsx`

### PR DESCRIPTION
**Pull request by gitstart-augur**
https://github.com/AugurProject/augur/issues/4172

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). ⭐️

 This link will always point to this PR's latest changes: https://peek.run/o9ph749

File: packages/augur-ui/src/modules/alerts/components/alerts-view.tsx [hunk 1]
```diff
@@ -1,107 +1,97 @@
...
+  clearAlerts,
+  toggleAlerts,
+  alertsVisible
+})=>{
+
+  React.useEffect(() =>{
+    if (alertsVisible) {
       alerts.forEach(alert => {
-        updateExistingAlert(alert.uniqueId, { ...alert, seen: true });
+        updateExistingAlert(alert.id, { ...alert, seen: true });
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/alerts/components/alerts-view.tsx]: can you use uniqueId here and add it to Alert type

**Review by phoebemirman — CHANGES_REQUESTED** [review]
please use uniqueId and add that to Alert type

File: packages/augur-ui/src/modules/alerts/components/alerts-view.tsx [hunk 2]
```diff
@@ -1,107 +1,97 @@
...
 import Styles from "modules/alerts/components/alerts-view.styles.less";
 import ToggleHeightStyles from "utils/toggle-height.styles.less";
 
 interface AlertsViewProps {
-  alerts: Array<any>;
-  updateExistingAlert: Function;
-  removeAlert: Function;
-  clearAlerts: Function;
-  toggleAlerts: Function;
+  alerts: Array<AlertType>;
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/alerts/components/alerts-view.tsx]: Can you do this array like AlertType[]

File: packages/augur-ui/src/modules/alerts/components/alerts-view.tsx [hunk 3]
```diff
@@ -1,107 +1,97 @@
-import React, { Component } from "react";
+import React from "react";
 import classNames from "classnames";
 
 import NullStateMessage from "modules/common/null-state-message";
 import Alert from "modules/alerts/components/alert";
-
+import {Alert as AlertType} from 'modules/types'
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/alerts/components/alerts-view.tsx:6]: Can you just import as Alert

Comment (gitstart-augur) [reply packages/augur-ui/src/modules/alerts/components/alerts-view.tsx:6]: We already used  the alert Keyword in Line 5 for a component import.

File: packages/augur-ui/src/modules/alerts/components/alert.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ import { ViewTransactionDetailsButton } from 'modules/common/buttons';
 import { convertUnixToFormattedDate } from 'utils/format-date';
 
 interface AlertProps {
-  id: string;
+  uniqueId: string;
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/alerts/components/alert.tsx:12]: can you keep id as well. it has both.

**Review by phoebemirman — CHANGES_REQUESTED** [review]
see comment

File: packages/augur-ui/src/modules/types.ts [hunk 5]
```diff
@@ -25,7 +25,7 @@ export enum SizeTypes {
 }
 
 export interface Alert {
-  id: string;
+  uniqueId: string;
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/types.ts:29]: keep id too

**Comment by bthaile** [comment]
split out transactions changes from alert changes

File: packages/augur-ui/src/modules/modal/transactions.tsx [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useState, useRef } from "react";
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/modal/transactions.tsx]: don't refactor transactions file in this PR

Comment (gitstart-augur) [reply packages/augur-ui/src/modules/modal/transactions.tsx]: Will revert this ASAP..

**Review by phoebemirman — CHANGES_REQUESTED** [review]
can you add id back to the Alert type

**Review by phoebemirman — APPROVED** [review]

**Pull request merged**